internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -0,0 +1,114 @@
+// Static implementation of TLS Configuration Store (no calls to S2Av2, Remote Signer Library, Certificate Verifier)
+package tls_config_store
+
+import (
+	"log"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"time"
+
+	_ "embed"
+)
+
+var (
+	//go:embed example_cert_key/client_cert.pem
+	clientCert []byte
+	//go:embed example_cert_key/server_cert.pem
+	serverCert []byte
+	//go:embed example_cert_key/client_key.pem
+	clientKey []byte
+	//go:embed example_cert_key/server_key.pem
+	serverKey []byte
+)
+
+func VerifyPeerCertificateFunc(instanceName string, pool *x509.CertPool) func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+	return func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+		if len(rawCerts) == 0 {
+			return fmt.Errorf("no certificate to verify")
+		}
+		cert, err := x509.ParseCertificate(rawCerts[0])
+		if err != nil {
+			return fmt.Errorf("x509.ParseCertificate(rawCerts[0]) returned error: %v", err)
+		}
+
+		opts := x509.VerifyOptions{
+			CurrentTime: time.Now(),
+			Roots:       pool,
+		}
+
+		if _, err = cert.Verify(opts); err != nil {
+			return err
+		}
+
+		if cert.Subject.CommonName != instanceName {
+			return fmt.Errorf("certificate had Common Name %q, expected %q", cert.Subject.CommonName, instanceName)
+		}
+		return nil
+	}
+}
+
+
+func GetTlsConfigurationForClient(id string) tls.Config {
+	// Static implementation. TODO : Call S2Av2 for these values.
+	min_version := uint16(tls.VersionTLS13)
+	max_version := uint16(tls.VersionTLS13)
+	var cipher_suites []uint16
+	var curve_preferences []tls.CurveID
+
+	// Static implementation. TODO : Call remote signer library for Private Key.
+	cert, err := tls.X509KeyPair(clientCert, clientKey)
+	if err != nil {
+		log.Fatalf("Failed to get client cert")
```

Comment:
Should we "Failed to generate X509Keypair"? Get seems a bit overloaded since it can also mean that the function is trying to obtain a certificate from somewhere rather that simply parsing the provided one.

Also, while we are at it, let's log the "err" return by tls.X509KeyPair method.

Reply:
Will do, thanks!